tests/test_views.py

```diff
@@ -155,37 +189,40 @@ def test_prepare_private_key(self):
         assert_equal(res.status_code, 200)
         assert_equal(res.request.GET['key'], self.link.key)
 
-    def test_cannot_access_registrations_or_forks_with_anon_key(self):
+    @mock.patch('framework.sessions.get_node_id')
+    def test_cannot_access_registrations_or_forks_with_anon_key(self, mock_get_node_id):
```

Comment:
These tests don't seem to cover all the possibly-hidden endpoints. Consider adding more for the other tabs that are hidden (Analytics, Settings, etc.) to ensure the correct error code is raised. This would help prevent regressions.